crt_portal/cts_forms/views.py

```diff
@@ -287,6 +314,41 @@ def get(self, request, id):
         output = serialize_data(report, request, id)
         contact_form = ContactEditForm(instance=report)
         details_form = ReportEditForm(instance=report)
+        return_url_args = request.GET.get('next', '')
+        index = request.GET.get('index', None)
+
+        # if we are filtering, allow the user to navigate through the
+        # filtered reports.
+        if return_url_args and index is not None:
+            querydict = QueryDict(return_url_args)
+            report_query, _ = report_filter(querydict)
+            sort = querydict.getlist('sort', ['-create_date'])
+            requested_query = report_query.order_by(*sort)
+            requested_ids = list(requested_query.values_list('id', flat=True))
+
+            index = int(index)
+            if report.id in requested_ids:
+                # override in case user input an invalid index
+                index = requested_ids.index(report.id)
+                output.update({
+                    'filter_current': index + 1,
+                })
+            else:
+                # this report is no longer in the filter, but we want
+                # to move backwards so that the previously next report
+                # becomes the actual next report.
+                index -= 1
+
+            previous_id = requested_ids[index - 1] if index > 0 else None
+            next_id = requested_ids[index + 1] if index < len(requested_ids) - 1 else None
+            next_query = urllib.parse.quote(return_url_args)
+            output.update({
+                'filter_count': report_query.count(),
+                'filter_previous': previous_id,
+                'filter_next': next_id,
+                'filter_previous_query': f'?next={next_query}&index={index - 1}',
+                'filter_next_query': f'?next={next_query}&index={index + 1}',
+            })
```

Comment:
I think you need to add something similar to the post method on line 366. The output returned by the post function does not have filter_count which is causing django to not render your navigation block

Reply:
Thanks! I abstracted away this functionality so I could apply it for the error edge cases. Should work better now (and the PR is cleaner as a result :tada: )